tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Collections.Immutable;
+using D2L.CodeStyle.Analyzers.Async.Generator;
+using D2L.CodeStyle.Annotations;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NUnit.Framework;
+using static D2L.CodeStyle.Analyzers.Async.Generator.SyntaxTransformer;
+
+namespace D2L.CodeStyle.Analyzers.Tests.Async.Generator;
+
+[TestFixture]
+internal sealed class AsyncToSyncMethodTransformerTests {
+	[Test]
+	public void BasicTask() {
+		var actual = Transform( @"[GenerateSync] async Task HelloAsync( int abc, string def ) { await Task.Delay( 3 ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Hello( int abc, string def ) {throw null;}", actual.Value.ToFullString() );

```

Comment:
There is a way to compare syntax trees, but `Assert.AreEqual` on strings has way better error messages when the test fails (shows you the first difference). Using the `SyntaxFactory.AreEquivalent` or `.IsEquivalentTo` results in opaque true/false style errors.

Reply:
The downside is that these assertions are sensitive to whitespace, e.g. `{throw null;}` because I didn't bother formatting that nicely. But I'm trying to be somewhat careful about mirror the source whitespace so that's maybe not so bad.

Reply:
I believe we do AreEquivalent in StaticDI and get the trees output in the message?

Reply:
I'm find with them being stringy though, given you do have exact goals.

Reply:
you just get true/false :'(

Reply:
K, in StaticDI we do: `Assert.Equal( expected, actual, SyntaxNodeComparer.Instance )` where SyntaxNodeComparer uses ` SyntaxFactory.AreEquivalent( x, y )` and get outputs.

Reply:
Oh right, what I disliked about that one is still problematic in here: it outputs the diff as a rendered string.

I used to have a version that asserted each node on the SyntaxNode structure recursively, accounting for whitespace.

The reason it mattered in static DI was that we hand SyntaxTrees to the compiler, and you can construct syntax trees that look good when rendered to a string but will cause the compiler to crash in odd ways.

Here, we will ultimately be handing strings to the compiler anyway (weird API). So I don't _really_ care about that stuff. Assert.AreEqual would probably work fine here, you're right.